collector/jsz.go

```diff
@@ -0,0 +1,354 @@
+			prometheus.BuildFQName(system, "server", "max_memory"),
+			"JetStream Max Memory",
+			serverLabels,
+			nil,
+		),
+		// jetstream_server_max_storage
+		maxStorage: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "server", "max_storage"),
+			"JetStream Max Storage",
+			serverLabels,
+			nil,
+		),
+		// jetstream_stream_total_messages
+		streamMessages: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "stream", "total_messages"),
+			"Total number of messages from a stream",
+			streamLabels,
+			nil,
+		),
+		// jetstream_stream_total_bytes
+		streamBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "stream", "total_bytes"),
+			"Total stored bytes from a stream",
+			streamLabels,
+			nil,
+		),
+		// jetstream_stream_state_last_seq
+		streamLastSeq: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "stream", "last_seq"),
+			"Last sequence from a stream",
+			streamLabels,
+			nil,
+		),
+		// jetstream_stream_consumer_count
+		streamConsumerCount: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "stream", "consumer_count"),
+			"Total number of consumers from a stream",
+			streamLabels,
+			nil,
+		),
+		// jetstream_consumer_delivered_consumer_seq
+		consumerDeliveredConsumerSeq: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "delivered_consumer_seq"),
+			"Latest sequence number of a stream consumer",
+			consumerLabels,
+			nil,
+		),
+		// jetstream_consumer_delivered_stream_seq
+		consumerDeliveredStreamSeq: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "delivered_stream_seq"),
+			"Latest sequence number of a stream",
+			consumerLabels,
+			nil,
+		),
+		// jetstream_consumer_num_ack_pending
+		consumerNumAckPending: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "num_ack_pending"),
+			"Number of pending acks from a consumer",
+			consumerLabels,
+			nil,
+		),
+		// jetstream_consumer_num_redelivered
+		consumerNumRedelivered: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "num_redelivered"),
+			"Number of redelivered messages from a consumer",
+			consumerLabels,
+			nil,
+		),
+		// jetstream_consumer_num_waiting
+		consumerNumWaiting: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "num_waiting"),
+			"Number of inflight fetch requests from a pull consumer",
+			consumerLabels,
+			nil,
+		),
+		// jetstream_consumer_num_pending
+		consumerNumPending: prometheus.NewDesc(
+			prometheus.BuildFQName(system, "consumer", "num_pending"),
+			"Number of pending messages from a consumer",
+			consumerLabels,
+			nil,
+		),
+	}
+
+	// Use the endpoint
+	nc.servers = make([]*CollectedServer, len(servers))
+	for i, s := range servers {
+		nc.servers[i] = &CollectedServer{
+			ID:  s.ID,
+			URL: s.URL,
+		}
+	}
+
+	return nc
+}
+
+// Describe shares the info description from a prometheus metric.
+func (nc *jszCollector) Describe(ch chan<- *prometheus.Desc) {
+	// Server state
```

Comment:
does it make sense to do a switch case over the endpoint here as well.
Basically saying if the endpoint says stream, don't describe consumer state.

Reply:
like one big switch case with fallthrough. start with 

switch endpoint {
      case consumer:
           ....
           fallthrough;
      case stream:
           .....
           fallthrough;
      case account:
           .....
           fallthrough;

}

Reply:
These ones are ok to leave as is, it is just the way that the exporter engine would show the help information but it would not be included if there are no relevant metrtics.

Reply:
This is only ok if the help is not used to list metrics. 
What I mean is if we set account, then maybe we shouldn't provide help for consumer.
Just to avoid prom suggesting metrics that will not exit.